sample_apps/simple_app/run_example.sh

```diff
@@ -0,0 +1,638 @@
+# run_example.sh - Driver script to build things needed to run sample-apps
+# ##############################################################################
+
+set -Eeuo pipefail
+
+# Setup script globals, to establish curdir and root path to Certifier code base
+Me=$(basename "$0")
+
+
+# Establish full dirpath where this script lives
+pushd "$(dirname "$0")" > /dev/null 2>&1
+EXAMPLE_DIR=$(pwd); export EXAMPLE_DIR
+popd > /dev/null 2>&1
+
+Cleanup="${EXAMPLE_DIR}/cleanup.sh"
+PROV_DIR="${EXAMPLE_DIR}/provisioning"
+
+# shellcheck disable=SC2046
+CERT_PROTO="$(dirname $(dirname "$EXAMPLE_DIR"))"; export CERT_PROTO
+CERT_UTILS="${CERT_PROTO}/utilities"
+ExName=$(basename "$EXAMPLE_DIR")
+
+MeMsg="Build and run example program ${ExName}"
+
+# Script global symbols to app-data dirs
+Client_app_data="./app1_data/"
+Srvr_app_data="./app2_data/"
+
+SimpleServer_PID=0
+AppAsServer_PID=0
+Found_step=0        # While looking thru Steps[] for user-specified fn-name
+
+# ###########################################################################
+# Set trap handlers for all errors. Needs -E (-o errtrace): Ensures that ERR
+# traps (see below) get inherited by functions, command substitutions, and
+# subshell environments
+# Ref: https://citizen428.net/blog/bash-error-handling-with-trap/
+#
+function cleanup() {
+    set +x
+    echo "${Me}: Failed command, ${BASH_COMMAND}, at line $(caller) while executing function ${This_fn}"
+
+    do_cleanup
+}
+
+# ###########################################################################
+# Calls the cleanup sub-script, which cleans up and returns 0 (always)
+# This avoids this script failing with non-zero $rc when stale stuff is cleaned up.
+function do_cleanup() {
+    ${Cleanup} $$
+}
+trap cleanup ERR
+
+# ##################################################################
+# Print help / usage
+# ##################################################################
+function usage() {
+
+   echo "You can use this script to ${MeMsg}.
+
+   - Setup and execute the example program.
+   - List the individual steps needed to setup the example program.
+   - Run individual step in sequence.
+"
+
+   # Computed elapsed hours, mins, seconds from total elapsed seconds
+   echo "Usage: $Me [--help | --list]"
+   echo "  To run the example program          : ./${Me}"
+   echo "  To setup the example program        : ./${Me} setup"
+   echo "  To run and test the example program : ./${Me} run_test"
+   echo "  To list the individual steps        : ./${Me} --list"
+   echo "  To run an individual step           : ./${Me} <step name>"
+}
+
+# ###########################################################################
+# Dump the list of functions, one for each step, that user can invoke.
+# Ref: https://www.freecodecamp.org/news/bash-array-how-to-declare-an-array-of-strings-in-a-bash-script/
+# ###########################################################################
+Steps=("show_env"
```

Comment:
each of the instructions in the instructions.md is now a "step" in this list of functions.

I will have to demo this to you to get a better understanding. `--help` should help.